craft_store/errors.py

```diff
@@ -0,0 +1,82 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2021 Canonical Ltd.
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License version 3 as published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Craft Store errors."""
+
+import dataclasses
+import logging
+
+import requests
+import urllib3  # type: ignore
+
+logger = logging.getLogger(__name__)
+
+
+class CraftStoreError(Exception):
+    """Base class error for craft-store."""
+
+
+@dataclasses.dataclass(repr=True)
```

Comment:
What's the benefit of using a dataclass here, instead of a more simple code like the following?

````
class NetworkError(CraftStoreError):

    def __init__(self, exception):
        self.exception = exception
        if exception.args and isinstance(exception.args[0], urllib3.exceptions.MaxRetryError):
            self.brief = "Maximum retries exceeded trying to reach the store."
        else:
            self.brief = str(exception)
        super().__init__(self.brief)

    def __str__(self) -> str:
        return self.brief
```

Reply:
This is done here: https://github.com/canonical/craft-store/pull/1/commits/9a3677adac8521a7271852034166ff469bd5cfa3

but moved brief initialization and instance var to the parent class to make it easier to interface with for a consumer of the API

Reply:
ack